docs/man/rpm-scriptlets.7.scd

```diff
@@ -0,0 +1,360 @@
+. *%triggerprein* of _new_, set off by _other_
+. *%pre* of _new_
+. (unpack _new_ files)
+. *%filetriggerin* of _other_ (high priority), set off by _new_
+. *%filetriggerin* of _new_ (high priority), set off by _other_
+. *%post* of _new_
+. *%triggerin* of _other_, set off by _new_
+. *%triggerin* of _new_, set off by _other_
+. *%filetriggerin* of _other_ (low priority), set off by _new_
+. *%filetriggerin* of _new_ (low priority), set off by _other_
+. *%triggerun* of _old_, set off by _other_
+. *%triggerun* of _other_, set off by _old_
+. *%filetriggerun* of _old_ (high priority), set off by _other_
+. *%filetriggerun* of _other_ (high priority), set off by _old_
+. *%preun* of _old_
+. *%filetriggerun* of _old_ (low priority), set off by _other_
+. *%filetriggerun* of _other_ (low priority), set off by _old_
+. (erase _old_ files)
+. *%filetriggerpostun* of _old_ (high priority), set off by _other_
+. *%filetriggerpostun* of _other_ (high priority), set off by _old_
+. *%postun* of _old_
+. *%filetriggerpostun* of _old_ (low priority), set off by _other_
+. *%filetriggerpostun* of _other_ (low priority), set off by _old_
+. *%posttrans* of _new_
+. *%postuntrans* of _old_
+. *%transfiletriggerpostun* of _other_, set off by _old_
+. *%transfiletriggerin* of _other_, set off by _new_
+
+# ENVIRONMENT
+*RPM_INSTALL_PREFIX*_N_
+	The _N_'th The install prefix in relocatable packages.
+
+*RPM_INSTALL_PREFIX*
+	The install prefix form in legacy relocatable packages.
+
+# EXIT STATUS
+A non-zero exit code from any scriptlet will be reflected as an error in the
+transaction result code. Additionally, non-zero exit from the various
+pre-scriptlets (noted above for each such scriptlet) stop further processing
+related to _that package_, not the whole transaction.
+
+Scriptlets be designed to always return with zero exit code, as RPM
+cannot undo or roll back a transaction.
+
+For example, an exit code of 1 from a *%pre* script prevents the install/update
+of that package, but the rest of the transaction will still take place,
+even if there are missing dependencies.
+
+# EXAMPLES
+*Example 1. Manage alternatives*
+
+```
+%post
+update-alternatives --install /usr/bin/frobber \\
+	frobber /usr/bin/frobber.myimpl 10
+
+%postun
+if [ $1 -eq 0 ] ; then
+	update-alternatives --remove frobber /usr/bin/frobber.myimpl 10
+fi
+```
+
+Note that these would go to the package owning the _/usr/bin/frobber.myimpl_
+binary, and similarly to all other alternatives to _/usr/bin/frobber_.
+
+*Example N. Manage the MIME database*
+
+```
+%transfiletriggerin -- /usr/share/mime
+/usr/bin/update-mime-database -n /usr/share/mime &> /dev/null ||:
+
+%transfiletriggerpostun -- /usr/share/mime
+/usr/bin/update-mime-database -n /usr/share/mime &> /dev/null ||:
+```
+
+Note that these would go to the package owning the
+_/usr/bin/update-mime-database_ binary.
+
+*Example N. Enable a systemd preset on upgrade from older version*
+
+```
+%triggerun -- my < 3.0-1
```

Comment:
This could (similarly to Example 1) also explain briefly *when* this actually executes, I think...